src/viam/sdk/common/exception.cpp

```diff
@@ -0,0 +1,55 @@
+#include <viam/sdk/common/exception.hpp>
+
+namespace viam {
+namespace sdk {
+
+ViamException::ViamException(const std::string& what)
+    : std::runtime_error("ViamException: " + what), code_(ViamErrorCode::Unknown){};
+
+ViamException::ViamException(const std::string& what, ViamErrorCode code)
+    : std::runtime_error("ViamException: " + what), code_(code){};
+
+ViamException::~ViamException() = default;
+
+std::error_code ViamException::code() const noexcept {
+    return code_;
+};
+
+struct ViamErrorCategory : std::error_category {
```

Comment:
Let's put this in the unnamed namespace - it doesn't need linkage as far as I can see.